File: application/src/main/java/com/study/platform/domain/post/usecase/UpdateStudyPostUseCase.java
package com.study.platform.domain.post.usecase;

import com.study.platform.domain.post.dto.request.StudyPostUpdateRequest;
import com.study.platform.domain.post.dto.response.StudyPostResponse;

import java.util.UUID;

public interface UpdateStudyPostUseCase {
    StudyPostResponse execute(UUID userId, UUID postId, StudyPostUpdateRequest request);
}

File: application/src/main/kotlin/com/study/platform/domain/post/usecase/UpdateStudyPostUseCase.kt
package com.study.platform.domain.post.usecase

import com.study.platform.domain.post.dto.request.StudyPostUpdateRequest
import com.study.platform.domain.post.dto.response.StudyPostResponse
import java.util.UUID

interface UpdateStudyPostUseCase {
    fun execute(userId: UUID, postId: UUID, request: StudyPostUpdateRequest): StudyPostResponse
}
